pkg/ccl/importccl/read_import_mysql.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package importccl
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	mysql "github.com/xwb1989/sqlparser"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/types"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+type mysqldumpReader struct {
+	conv     rowConverter
+	debugRow func(tree.Datums)
+}
+
+var _ inputConverter = &mysqldumpReader{}
+
+func newMysqldumpReader(
+	kvCh chan kvBatch, table *sqlbase.TableDescriptor,
+) (*mysqldumpReader, error) {
+	conv, err := newRowConverter(table, kvCh)
+	if err != nil {
+		return nil, err
+	}
+	return &mysqldumpReader{conv: *conv}, err
+}
+
+func (m *mysqldumpReader) start(ctx ctxgroup.Group) {
+}
+
+func (m *mysqldumpReader) inputFinished(ctx context.Context) {
+	close(m.conv.kvCh)
+}
+
+func (m *mysqldumpReader) readFile(
+	ctx context.Context, input io.Reader, inputIdx int32, inputName string, progressFn progressFn,
+) error {
+	evalCtx := &tree.EvalContext{SessionData: &sessiondata.SessionData{Location: time.UTC}}
```

Comment:
is this the safest way to get an evalctx? it seems real brittle since other required members of the struct could be added in the future and cause breakage.

Reply:
yeah, this was cribbed from CSV, but there's actually an evalCtx generator up on the flowCtx we can plumb though.

Reply:
i'm fine not changing this now if it's in multiple places. we can make an issue we'll never fix for it too.

Reply:
https://github.com/cockroachdb/cockroach/pull/26057

Reply:
(i'll rebase this one on top of #26057)